src/lib.rs

```diff
@@ -197,6 +197,35 @@ impl Default for Scissor {
     }
 }
 
+impl Scissor {
+    /// Returns the bounding rect if the scissor clip if it's an untransformed rectangular clip
+    fn as_rect(&self, canvas_width: f32, canvas_height: f32) -> Option<Rect> {
+        let extent = match self.extent {
+            Some(extent) => extent,
+            None => return Some(Rect::new(0., 0., canvas_width, canvas_height)),
+        };
```

Comment:
let extent = self.extent.unwrap_or_else(|| Rect::new(0., 0., canvas_width, canvas_height))

Reply:
Hm, that's not quite the same though. I think if self.extent is None, no scissor is set and the code returns, instead of continuing.

I could change this to continue, then it would look like this:

```patch
diff --git a/src/lib.rs b/src/lib.rs
index 19b479a..3fcf4c6 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -200,10 +200,13 @@ impl Default for Scissor {
 impl Scissor {
     /// Returns the bounding rect if the scissor clip if it's an untransformed rectangular clip
     fn as_rect(&self, canvas_width: f32, canvas_height: f32) -> Option<Rect> {
-        let extent = match self.extent {
-            Some(extent) => extent,
-            None => return Some(Rect::new(0., 0., canvas_width, canvas_height)),
-        };
+        let mut scissor_x = self.transform[4];
+        let mut scissor_y = self.transform[5];
+        let extent = self.extent.unwrap_or_else(|| {
+            scissor_x = canvas_width / 2.;
+            scissor_y = canvas_height / 2.;
+            [canvas_width / 2., canvas_height / 2.]
+        });

         // Abort if we're skewing (usually doesn't happen)
         if self.transform[1] != 0.0 || self.transform[2] != 0.0 {
@@ -218,8 +221,8 @@ impl Scissor {
         let half_width = extent[0];
         let half_height = extent[1];
         Some(Rect::new(
-            self.transform[4] - half_width,
-            self.transform[5] - half_height,
+            scissor_x - half_width,
+            scissor_y - half_height,
             half_width * 2.0,
             half_height * 2.0,
         ))
```

I think the early return is better than the mutation in the else.